src/tests/ftest/util/general_utils.py

```diff
@@ -707,21 +755,10 @@ def stop_processes(hosts, pattern, verbose=True, timeout=60, added_filter=None,
 
     if hosts is not None:
         if dump_ult_stacks is True and "daos_engine" in pattern:
-            commands_part1 = [
-                "rc=0",
-                "if " + ps_cmd,
-                "then rc=1",
-                "sudo pkill --signal USR2 {}".format(pattern),
-                # leave time for ABT info/stacks dump vs xstream/pool/ULT number
-		"sleep 20",
-                "fi",
-                "exit $rc",
-            ]
-            result = pcmd(hosts, "; ".join(commands_part1), verbose, timeout,
-                          None)
+            result = dump_engines_stacks(hosts, verbose, timeout, added_filter)
```

Comment:
Not that this is new, but capturing the return value from `dump_engine_stacks()` gets lost when on line 777 `result` is redefined with return value of `pcmd()`.  There really isn't a clean way to combine the two results.  Hypothetically calling dump_engine_stacks() could result in a failure, but the pkill of the various commands does not and there would be no indication in the return value that something went wrong.  

With the latest changes, we don't go through this method to dump the ULT stacks, so for the two remaining cases that call this method it probably makes sense to remove the `dump_ult_stacks=False` option and change to code from:
```
result = stop_processes(hosts, pattern, verbose=True, timeout=60, added_filter=None, dump_ult_stacks=True)
```
to:
```
result1 = dump_engines_stacks(hosts, verbose, timeout, added_filter)
result2 = stop_processes(hosts, pattern, verbose=True, timeout=60, added_filter=None)
```
Then the calling code can decide whether to do something - like fail the test - if `result1` contains a failed result.

Reply:
Well, I am not sure this is a good idea as it will prevent to warn new users of the stop_processes() to think/forget about dumping ULTs stacks or not

Reply:
I understand that concern, however not reporting a failing command status is also a problem as it will hide an issue - the ULT dump command could be failing, but we wouldn't know about it until we had another failure that is reported.  

The problem is the dictionary returned, e.g. `result`,  cannot be combined in a meaningful way with another `result`.  So we cannot call `dump_engines_stacks` and then usefully combine the result of the stopping the processes.  Even if we could this would most likely still require the caller to decipher the result and decide what to do with multiple permutations (dump fails but stop passes, dump passes but stop fails, and both dump and stop fail).  Since the action based upon these outcomes are caller dependent I would argue we're back to calling each individually:
```
result1 = dump_engines_stacks(hosts, verbose, timeout, added_filter)
result2 = stop_processes(hosts, pattern, verbose=True, timeout=60, added_filter=None)
```
Where the caller decides what to do with each unsuccessful result.

I could see a case for a simplified `stop_engines()` method that does something like:
```
def stop_engines(hosts, pattern, verbose=True, timeout=60, added_filter=None):
    result = dump_engines_stacks(hosts, verbose, timeout, added_filter)
    if 0 in result and len(result) == 1:
        result = stop_processes(hosts, pattern, verbose, timeout, added_filter)
    return result
```
This would enable normal tests actions of stopping the engines to call a single method, but cleanup code where we would want to report `dump_engines_stacks` failures but still call `stop_processes` (the version that doesn't include dumping ULT stacks) and report if it failed as well.

This a long-winded way of saying the functions should be more succinct and focused.

Reply:
Well, I understand it is always a bad idea to ignore error-cases, but in this particular one, dumping of ULTs stacks is already an enhancement as a best-effort and detecting its failure, along with handling it, is a whole task on its own and that could only be worked-on, as an add-on patch, when [PR-374](https://github.com/pmodels/argobots/pull/374) for issue [#372 ](https://github.com/pmodels/argobots/issues/372) at Argobots will be present in the ABT version we use.
So, I propose that we land this PR as is , if nothing else needs to be changed..., and I will later push a PR to handle ULTs stacks dump failure detection/reporting.

Reply:
Ok, how about this as a disagree and commit approach?  Let's keep the behavior of `stop_processes()` as you have it, but pull out the code that actually stops the processes via a pattern into a new method so that any future code that may want to check the status of either action can do so.

So `stop_processes()` would look something like this:
```
def stop_processes(hosts, pattern, verbose=True, timeout=60, added_filter=None,
                   dump_ult_stacks=False):
    """Stop the processes on each hosts that match the pattern.
    ...
    """
    log = getLogger()
    if dump_ult_stacks is True and "daos_engine" in pattern:
        log.info("First dumping ULT stacks, then Killing any processes on %s "
                 "that match: %s", hosts, pattern)
        dump_engines_stacks(hosts, verbose, timeout, added_filter)
    return stop_process_pattern(hosts, pattern, verbose, timeout, added_filter)
```

And the code pulled out of `stop_processes()` would look something like this:
```
def stop_process_pattern(hosts, pattern, verbose=True, timeout=60, added_filter=None):
    """Stop the processes on each hosts that match the pattern.
    ...
    """
    result = {}
    log = getLogger()
    log.info("Killing any processes on %s that match: %s", hosts, pattern)
    if hosts is not None:
        if added_filter:
            ps_cmd = "/usr/bin/ps xa | grep -E {} | grep -vE {}".format(pattern, added_filter)
        else:
            ps_cmd = "/usr/bin/pgrep --list-full {}".format(pattern)
        commands = [
            "rc=0",
            "if " + ps_cmd,
            "then rc=1",
            "sudo /usr/bin/pkill {}".format(pattern),
            "sleep 5",
            "if " + ps_cmd,
            "then sudo /usr/bin/pkill --signal ABRT {}".format(pattern),
            "sleep 1",
            "if " + ps_cmd,
            "then sudo /usr/bin/pkill --signal KILL {}".format(pattern),
            "fi",
            "fi",
            "fi",
            "exit $rc",
        ]
        result = pcmd(hosts, "; ".join(commands), verbose, timeout, None)
    return result
```

Reply:
From what I understand, there is no modification with the patch proposed by @bfaccini : in the legacy code command_part2 was also lost: line 740 was overloading the assignment of the line 720. FMHO, trying to not lost the result of this first command is an enhancement which should be done in a new JIRA ticket.
It should also be evaluated if it is still relevant to call `dump_engine_stack `in the `stop_processes` function: it is already called in the different epilogue of the avocado framework.